### PR TITLE
fix: mixpanel use USD amounts instead of user-currency amounts

### LINKED ISSUE #4866: Always use USD when passing fiatAmounts to MixPanel
### Overview

On June 28th product noticed a significant number of trades in Mixpanel that showed fiatAmounts that did not match the actual value (ie. [this users' trade success events](https://mixpanel.com/project/2933919/view/3436921/app/profile#distinct_id=1890416ab91341-0497729221002e-26031d51-e1000-1890416ab92d12) where a 27.719 USDT trade has a fiatAmount of 415087.9367675082082368).

The cause of this appears to be that the Mixpanel data currently does not account for user's currency preferences. In the example above, the user is based in Indonesia, and the value of 27.5 USD is about 415k indonesian rupiahs.
image

### References and additional details

Can find examples of this by scrolling through the Trade Success events in mixpanel, filtered by highest fiatAmount
https://mixpanel.com/project/2933919/view/3436921/app/events#s46KPDpgz6Yw

If you know your userID in mixpanel you can test this by adjusting your fiat currency and triggering the trade confirm event.

### Acceptance Criteria

AC:
1. fiatAmount properties in Mixpanel (including donationFiatAmount) are always in USD
2. If this is not feasible, another much less desirable option is to add a property for Currency; this would at least enable us to filter out the noise pkus we could do rough conversions in mixpanel for most common currencies

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR:

- brings in new USD selector in addition to the current user-currency ones for donation and sell amounts, both in old and new swapper
- sends the data to mixpanel in USD both for donations and sell amounts *in addition to* the data in user-currency
- sends the `selectedCurrency` in

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #4866

## Risk

If we got this wrong, tracking will be unreliable - test accordingly

## Testing

- Make a few trades, both with and without donations, both with USD and another currency
- Ensure the fiat amounts (sell asset/donation) are correctly passed in in USD and selected currency, alongside the selected currency

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

### CoW - no donation

Screenshot 2023-07-05 at 17 38 06
image
image
image

### 1inch - with donation

image
image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4892** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4892" target="_blank">Graphite</a> 
    * **PR #4893** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4893" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4893?utm_source=stack-comment).

File: src/components/Trade/TradeConfirm/TradeConfirm.tsx [hunk 1]
```diff
@@ -273,10 +277,10 @@ export const TradeConfirm = () => {
     return {
       buyAsset: compositeBuyAsset,
       sellAsset: compositeSellAsset,
-      fiatAmount: sellAmountBeforeFeesUserCurrency,
+      amountUSD: sellAmountBeforeFeesUsd,
```
Comment (gomesalexandre) [inline src/components/Trade/TradeConfirm/TradeConfirm.tsx:280]:
```suggestion
      amountUsd: sellAmountBeforeFeesUsd,
```

File: src/state/slices/tradeQuoteSlice/selectors.ts [hunk 2]
```diff
@@ -344,7 +344,16 @@ export const selectSellAmountUserCurrency = createSelector(
   },
 )
 
-export const selectDonationAmountFiat = createSelector(
+export const selectSellAmountUsd = createSelector(
```
Comment (gomesalexandre) [inline src/state/slices/tradeQuoteSlice/selectors.ts:347]: can consume the existing selector and div by the user currency to usd rate

Comment (gomesalexandre) [reply src/state/slices/tradeQuoteSlice/selectors.ts:347]: or actually meh, potato potato

File: src/state/slices/tradeQuoteSlice/selectors.ts [hunk 3]
```diff
@@ -355,6 +364,17 @@ export const selectDonationAmountFiat = createSelector(
   },
 )
 
+export const selectDonationAmountUsd = createSelector(
```
Comment (gomesalexandre) [inline src/state/slices/tradeQuoteSlice/selectors.ts:367]: This one may actually make sense to consume the existing selector not to duplicate complexity

File: src/components/Trade/TradeConfirm/TradeConfirm.tsx (hunk 1 above)
Comment (gomesalexandre) [reply src/components/Trade/TradeConfirm/TradeConfirm.tsx]: 3ec6925cd

File: src/state/slices/tradeQuoteSlice/selectors.ts (hunk 3 above)
Comment (gomesalexandre) [reply src/state/slices/tradeQuoteSlice/selectors.ts:367]: 3ec6925cd

**Comment by firebomb1** [comment]
As discussed in DM and from the closed PR, tracking the user currency symbol and the user currency amounts since we have them would be great (potentially allow regional reporting in the user's currency without having to backtrack rates).

**Comment by firebomb1** [comment]
LGTM! Thanks for these changes!

**Review by kaladinlight — APPROVED** [review]

**Comment by gomesalexandre** [comment]
@gomesalexandre started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4893).

**Comment by gomesalexandre** [comment]
[Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4893) couldn't merge this PR because it had conflicts with the trunk branch.

**Pull request merged**